src/main/kotlin/com/coder/gateway/views/CoderGatewayRecentWorkspaceConnectionsView.kt

```diff
@@ -174,14 +174,15 @@ class CoderGatewayRecentWorkspaceConnectionsView(private val setContentCallback:
                         val workspaceWithAgent = deployment?.items?.firstOrNull { it.workspace.name == workspaceName }
                         val status =
                             if (deploymentError != null) {
-                                Pair(UIUtil.getErrorForeground(), deploymentError)
+                                Triple(UIUtil.getErrorForeground(), deploymentError, UIUtil.getBalloonErrorIcon())
                             } else if (workspaceWithAgent != null) {
-                                Pair(
+                                Triple(
                                     workspaceWithAgent.status.statusColor(),
                                     workspaceWithAgent.status.description,
+                                    null
```

Comment:
Another idle thought so feel free to ignore!  It could make sense to use this block for setting the workpsace's idling icon, together with its status:

```suggestion
                                val inLoadingState = listOf(WorkspaceStatus.STARTING, WorkspaceStatus.CANCELING, WorkspaceStatus.DELETING, WorkspaceStatus.STOPPING).contains(workspaceWithAgent.workspace.latestBuild.status)
                                Triple(
                                    workspaceWithAgent.status.statusColor(),
                                    workspaceWithAgent.status.description,
                                    if (isLoadingState) AnimatedIcon.Default() else null
                                )
```

Reply:
Yeah I totally agree, consolidating determining what icon to display to one place in the file is much better.